### PR TITLE
Reimplement ISPN-9976 test for Elastic Search availability 

### PR DESCRIPTION
**Pull request by anistor**
https://issues.jboss.org/browse/ISPN-9976

### CODE REVIEW

File: query/src/main/java/org/infinispan/query/dsl/embedded/impl/HibernateSearchPropertyHelper.java [hunk 1]
```diff
@@ -149,6 +142,15 @@ private Object convertToPropertyType(String value, FieldBridge bridge) {
       }
    }
 
+   private boolean isElasticsearchDateBridge(FieldBridge bridge) {
+      try {
+         return bridge instanceof org.hibernate.search.elasticsearch.bridge.builtin.impl.ElasticsearchDateBridge;
+      } catch (Exception e) {
+         // this happens because Elastic search could be missing from classpath, being an optional dependency
+         return false;
```
Comment (ryanemerson) [inline query/src/main/java/org/infinispan/query/dsl/embedded/impl/HibernateSearchPropertyHelper.java]: I'm not sure why this is preferable to having the check in the constructor? With the constructor approach, we only have to catch a Exception once. Whereas now an exception is potentially thrown per invocation of `convertToPropertyType`

Comment (anistor) [reply query/src/main/java/org/infinispan/query/dsl/embedded/impl/HibernateSearchPropertyHelper.java]: I'm fine with any approach that does not require the ClassLoader parameter to those constructors. The change was a bit disruptive. So to mitigate the 'per invocation' we can add the flag back.

**Comment by anistor** [comment]
Updated.

**Comment by anistor** [comment]
Seems like CNFE happens, so this back to the drawing board.

**Comment by anistor** [comment]
Redone, even simpler.

**Pull request merged**